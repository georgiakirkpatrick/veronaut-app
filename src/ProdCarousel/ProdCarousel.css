.ProdCarousel {
    width: 100%;
    overflow: scroll;
}

.ProdCarousel__items {
    display: flex;
    flex-direction: row;
}

.ProdCarousel__items > li {
    padding: 0 10px 0 0;
}

.ProdCarousel__items > li > img {
    width: 80vw;
}

@media screen and (min-width: 768px) {
    .ProdCarousel__items > li > img {
        width: 40vw;
    }
}

@media screen and (min-width: 1024px) {
    .ProdCarousel {
        width: 40%;
    }

    .ProdCarousel__items {
        flex-direction: column;
        align-items: center;
    }

    .ProdCarousel__items > li > img {
        width: auto;
        height: 70vh;
    }
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .ProdCarousel {
        width: 40%;
    }

    .ProdCarousel__items > li > img {
        height: 30vh;
    }
}

@media screen and (min-width: 1440px) {
    .ProdCarousel {
        width: 40%;
    }

    .ProdCarousel__items > li > img {
        height: 50vh;
    }
}

@media screen and (min-width: 2560px) {
    .ProdCarousel {
        width: 50%;
    }

    .ProdCarousel__items > li > img {
        height: 40vh;
    }
}