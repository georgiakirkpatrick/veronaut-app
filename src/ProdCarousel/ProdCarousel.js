import React from 'react'
import './ProdCarousel.css'

const ProdCarousel = props => {
    const {
        colorArray,
        id,
        productName,
        selectedColorId
    } = props
 
    const makeColorImages = color => {
        const primaryImage = color.imageArray.find(image => image.primaryImage === true)
        const allButPrimary = color.imageArray.filter(image => image.imageId !== primaryImage.imageId)

        return [
            primaryImage,
            ...allButPrimary
        ]
    }

    const selectedColor = colorArray.filter(color => color.colorId === selectedColorId)[0]
    const imagesForColor = makeColorImages(selectedColor)
    const formattedImages = imagesForColor.map(image => {
        return <li key={'li-' + image.imageId}>
            <img
                src={image.imageUrl}
                alt={`${productName} ${selectedColor.colorEngName} ${image.imageId}`}
            />
        </li>
    })

    return (
        <div 
            id={id}
            className='ProdCarousel'
        >
            <ul className='ProdCarousel__items'>
                {formattedImages}
            </ul>
        </div>
    )
}

ProdCarousel.defaultProps = {
    id: '',
    colorArray: [
        {
            colorDescId: 1,
            colorEngName: '',
            colorId: 1,
            imageArray: [
                {
                    imageId: 1,
                    imageUrl: '',
                    primaryImage: true,
                }
            ],
            swatchUrl: ''
        }
    ],
    selectedColorId: 1,
    productName: ''
}

export default ProdCarousel