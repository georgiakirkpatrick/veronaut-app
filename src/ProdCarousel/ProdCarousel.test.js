import React from 'react'
import ReactDOM from 'react-dom'
import ProdCarousel from './ProdCarousel'
import renderer from 'react-test-renderer'

describe ('ProdCarousel', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ProdCarousel />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ProdCarousel />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})