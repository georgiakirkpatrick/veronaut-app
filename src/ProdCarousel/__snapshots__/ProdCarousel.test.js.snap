// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProdCarousel renders the UI as expected 1`] = `
<div
  className="ProdCarousel"
  id=""
>
  <ul
    className="ProdCarousel__items"
  >
    <li>
      <img
        alt="  1"
        src=""
      />
    </li>
  </ul>
</div>
`;
